### PR TITLE
VMware: Specify content library

### PR DESCRIPTION
**Pull request by hsitin**
1) specify the name of the content library from where the template needs to be picked.

SUMMARY
Added the functionality to be able to specify the name of the content library from where the template needs to be picked from. The existing functionality will prevail if the field content_library
is not specified.

ISSUE TYPE
Feature Pull Request

COMPONENT NAME
vmware_content_deploy_template.py

ADDITIONAL INFORMATION
Currently, if we have templates with the same name in multiple content libraries then the first template that matches the specified template name is returned. Being able to specify the content library name with the template name gives more control for choosing the template.

###### If the template or the content library specified does not exist
fatal: [localhost -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "cluster": null,
            "content_library": "<content_library_name>",
            "datacenter": "<datacenter name>",
            "datastore": "<datastore_name>",
            "folder": "<folder_name>",
            "host": "<host_name>",
            "hostname": "<vcenter_name>",
            "name": "<virtual_machine_name>",
            "password": "<password>",
            "protocol": "https",
            "resource_pool": null,
            "state": "poweredon",
            "template": "<template_name>",
            "username": "<user_name>",
            "validate_certs": false
        }
    },
    "msg": "Failed to find the library Item <template_name> in content library <content_library_name>"
}

### CODE REVIEW

**Comment by jimi-c** [comment]
Thank you very much for your interest in Ansible. 

We're sorry for the inconvenience, however this module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/vmware. Please re-submit this issue there so it can be properly reviewed. 

If you have further questions please stop by IRC or the mailing list: 

   * https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
   * https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code
   * IRC: #ansible or #ansible-devel on irc.freenode.net

Thanks!

**Comment by jimi-c** [comment]
My mistake, I didn't see that this was a backport for 2.9 until just after closing it. Reopened now.

**Comment by hsitin** [comment]
Thanks @jimi-c . I would like to know what needs to be done from my end to get this PR merged.

**Comment by sivel** [comment]
I believe this is the upstream PR: https://github.com/ansible-collections/vmware/pull/188

@hsitin backports are merged by release managers shortly before creating a new release. It will be evaluated and potentially merged at a later date.

**Comment by hsitin** [comment]
@sivel That sounds perfect. Thanks.

**Comment by Akasurde** [comment]
@hsitin This is not a possible candidate for backport. Since this is a adding a new parameter in the module.

**Review by Akasurde — CHANGES_REQUESTED** [review]
As above

**Comment by hsitin** [comment]
@Akasurde Got it. As this PR is already merged in "ansible-collections/vmware", the new parameter will be available once ansible 2.10 is released ? Also, should I close this PR ?

**Comment by Akasurde** [comment]
With the collection, you don't need to wait for Ansible 2.10 release.

You can either
1. wait for the latest release of VMware collection with the fix or 
2. build devel version of the collection using the following steps
	- git clone https://github.com/ansible-collections/vmware
	- cd vmware
	- echo "version: temp" >> galaxy.yml
	- ansible-galaxy collection build .
	- ansible-galaxy collection install ./community-vmware-temp.tar.gz

I will close this issue for you. Thanks for your contribution.

**Pull request closed without merging**
